generators/lib/distros/octane.js

```diff
@@ -31,21 +32,29 @@ function init() {
     done();
   };
 
-  module.modifyComposer = function(yo, options, localComposer, isNewProject, done) {
+  module.loadComposer = function(yo, options) {
+    var composer = {};
+    var file = yo.templatePath('drupal/drupal/' + options.drupalDistroRelease + '/composer.json');
+    if (gadget.fsExistsSync(file)) {
+      composer = yo.fs.readJSON(file);
+      // Octane gets it's version of Drupal Core from Lightning.
+      // To minimize conflicts with the rest of the generator process, we defer
+      // removing this until the last time Octane can reasonably impose this logic.
+      delete composer.require['drupal/core'];
+    }
+    return composer;
+  }
+
+  module.modifyComposer = function(yo, options, localComposer, isNewProject, done, cb) {
```

Comment:
cb is a standard node convention for a callback, which is how results will be delivered.

One way we could reduce the argument list is to attach more values as "semi-global" state to the yo object (`this` in index.js). I'm not sure why we avoid that so much.

Improving the comments or making a change sufficient to simplify the parameters here seems like a really good idea, but I'd rather do that refactor in a followup.